molecule/testinfra/staging/common/test_grsecurity.py

```diff
@@ -192,3 +192,15 @@ def test_wireless_disabled_in_kernel_config(host, kernel_opts):
 
     line = "# CONFIG_{} is not set".format(kernel_opts)
     assert line in kernel_config
+
+
+def test_mds_mitigations_and_smt_disabled(host):
+    """
+    Ensure that full mitigations are in place for MDS
+    see https://www.kernel.org/doc/html/latest/admin-guide/hw-vuln/mds.html
+    """
+
+    grub_config_path = "/boot/grub/grub.cfg"
+    grub_config = host.file(grub_config_path)
+
+    assert grub_config.contains("mds=full,nosmt")
```

Comment:
Ideally we'd check `/sys/devices/system/cpu/vulnerabilities/mds`, but checking the mitigations in VMs won't give us an accurate picture about the mitigations on hardware, so fine as-is.

Reply:
Similarly, we'll continue to run the spectre/meltdown checker script manually during QA. If we added it to the test run here, we'd still see failures in VMs. Worth revisiting after the release if manual script runs are showing green on hardware.